app/services/api/base_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Api
+  class BaseService
+    include ServicePattern
+    include Config
+
+    def initialize(params = {})
+      @params = params
+    end
+
+  private
+
+    attr_reader :params
+
+    def pagination_per_page
+      params.fetch(:per_page, PAGINATION_PER_PAGE).to_i
+    end
+
+    def page
+      params.fetch(:page, 1).to_i
+    end
+
+    def sort_by
+      params.fetch(:sort_by, SORT_BY)
+    end
+
+    def since
+      params.fetch(:since, Date.new).to_date
```

Comment:
Does it make sense to default to returning only those trainees created since the start of the current day? Not sure but I wonder whether just skipping the filter might be the least surprising thing (if we default to the current cycle then we won't get too many results).